pkg/registry/storage/csinode/strategy.go

```diff
@@ -69,8 +74,20 @@ func (csiNodeStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Ob
 func (csiNodeStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
 	newCSINodeObj := obj.(*storage.CSINode)
 	oldCSINodeObj := old.(*storage.CSINode)
-	errorList := validation.ValidateCSINode(newCSINodeObj)
-	return append(errorList, validation.ValidateCSINodeUpdate(newCSINodeObj, oldCSINodeObj)...)
+	// in 1.21 on update, set AllowLongNodeID to true only if the old object already has a long node ID
+	// in 1.22 on update, set AllowLongNodeID=true
```

Comment:
make sure there's a tracking issue open to change this in 1.22

Reply:
Thanks, I created https://github.com/kubernetes/kubernetes/issues/99981 for it.